Packs/Base/Scripts/GetIncidentsByQuery/GetIncidentsByQuery.py

```diff
@@ -18,23 +17,29 @@ def parse_datetime(datetime_str):
 
 
 def parse_relative_time(datetime_str):
-    try:
-        res = re.search("([0-9]+) (minutes|hours|days|weeks|months|years) ago", datetime_str)
-        if res:
-            number = int(res.group(1))
-            unit = res.group(2)
-            if unit == 'years':
-                unit = 'days'
-                number *= 365
-            elif unit == 'months':
-                number *= 43800
-                unit = 'minutes'
-            kargs = {}
-            kargs[unit] = int(number)
-            result = datetime.now() - timedelta(**kargs)
-            return result
-    except Exception:
-        return None
+    if datetime_str:
+        datetime_str = datetime_str.lower()
+        try:
+            res = re.search("([0-9]+) (minute|minutes|hour|hours|day|days|week|weeks|month|months|year|years) ago",
+                            datetime_str)
+            if res:
+                number = int(res.group(1))
+                unit = res.group(2)
+                if unit in ['minute', 'hour', 'day', 'week', 'month', 'year']:
+                    unit += "s"
+                if unit == 'years':
+                    unit = 'days'
+                    number *= 365
+                elif unit == 'months':
+                    number *= 43800
+                    unit = 'minutes'
+
+                kargs = {}
+                kargs[unit] = int(number)
+                result = datetime.now() - timedelta(**kargs)
+                return result
+        except Exception:
+            return None
```

Comment:
Perhaps it's better to return an error here? might be confusing

Reply:
I handle this as log error... I don't want to return error - not breaking things